src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Simple.cs

```diff
@@ -513,6 +520,82 @@ public void OperationsOnUnconnectedClient()
         }
 
         [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        // [ActiveIssue(2640)]
+        public async Task CancelTokenOff_Server_ReadWriteCancelledToken_Throws_OperationCanceledException()
+        {
+            using (NamedPipePair pair = CreateNamedPipePair())
+            {
+                NamedPipeServerStream server = pair.serverStream;
+                byte[] buffer = new byte[] { 0, 0, 0, 0 };
+
+                pair.Connect();
+
+                if (server.CanRead)
+                {
+                    Task serverReadToken = server.ReadAsync(buffer, 0, buffer.Length, CancellationToken.None);
+                    unsafe
+                    {
+                        Assert.True(CancelIoEx(server.SafePipeHandle, null), "Outer cancellation failed");
```

Comment:
Nevermind; I misread this code... I missed that you were asserting on the result of CancelIoEx... I thought you were asserting that server.SafePipeHandle == null.